src/runtime/button-api.js

```diff
@@ -60,10 +72,14 @@ export class ButtonApi {
 
   /**
    * @param {!../model/doc.Doc} doc
+   * @param {!function(!../api/client-event-manager-api.ClientEvent)} logEventFun
    */
-  constructor(doc) {
+  constructor(doc, logEventFun) {
```

Comment:
if constructor was runtime or event manager, then the click event listener would just send event to it directly.